packages/dai-plugin-mcd/src/CdpTypeService.js

```diff
@@ -32,4 +36,86 @@ export default class CdpTypeService extends PublicService {
     assert(types.length <= 1, `${label} matches more than one cdp type`);
     assert(types.length > 0, `${label} matches no cdp type`);
   }
+
+  useMulticall(watcher) {
```

Comment:
I'm a bit hesitant about this API. CdpTypeService shouldn't know much about multicall, and callers of the API should definitely not know about multicall.

I can see mutlicall being used elsewhere in mcd, so would it make sense to pass multicall as a configuration to the plugin itself? That was multiple services should be able to access the same instance, and we'd be able to create some sort of generic 'watcher' service that multiple services could rely on.

Reply:
I agree that makes way more sense conceptually. The tricky thing is, I was running into a circular dependency. In the CDP portal, multicall gets its RPC URL from the Maker instance. So we don't have a multicall instance yet when we configure Maker. So I went with this cheap hack just to get through illustrating the basic concept. 

Will have to keep mulling over how to untangle that. On the one hand, we could have a "watcher" service instantiate multicall during Maker setup. 

On the other hand, that's not a good design for the long-term vision of multicall, where an app has a multicall singleton that several different libraries plug into. This would require multicall and Maker to handle their RPC URL setup independently. Which would lead to duplication of effort around managing different providers, switching networks, etc.

Now that I've written that out, it's probably more practical to have the watcher service own the multicall instance at this time. What do you all think?

Reply:
^ @michaelelliot